linode/apl-core
# feat: add thanos for prom HA

This PR adds Thanos for Prometheus HA and includes:
- add Thanos Helm chart
- configure Prometheus to use Thanos sidecars
- create `objstore.yml` for OBJ configuration based on configured OBJ provider
- add thanos-query datasource to all Grafana's
- add Thanos Grafana dashboards to Platform Grafana

## Testing

- Configure OBJ with a bucket for thanos
- Activate Thanos app
- Check to see if the thanos sidecar is added to the Prometheus pod
- Check (using the Thanos query UI) to see if the sidecar endpoint status is up
- Check if (this can take some time) if the Store endpoint (only one) has status Up
- Set the replica's for prometheus to `2`
- Check to see if the second sidecar endpoint comes up
- Check to see if all the announced LabelSets (for both replica's) are shown

You should now see the following:

![Screenshot 2024-09-25 at 14 41 17](https://github.com/user-attachments/assets/461b18ff-3a7c-4217-8791-e4339fd66b17)

## Checklist
